esmvalcore/preprocessor/_regrid.py

```diff
@@ -306,6 +307,48 @@ def _attempt_irregular_regridding(cube, scheme):
     return False
 
 
+def extract_location(cube, location, scheme):
+    """Extract a point using a location name, with interpolation.
+
+    Extracts a single location point from a cube, according
+    to the interpolation scheme `scheme`.
+
+    The function just retrieves the coordinates of the location and then calls
+    the `extract_point` preprocessor.
+
+    It can be use to locate cities and villages, but also mountains or other
+    geographical locations.
+
+    Parameters
+    ----------
+    cube : cube
+        The source cube to extract a point from.
+
+    location : str
+        The reference location.
+
+    scheme : str
+        The interpolation scheme. 'linear' or 'nearest'. No default.
+
+    Returns
+    -------
+    Returns a cube with the extracted point, and with adjusted
+    latitude and longitude coordinates.
+
+    Raises
+    ------
+    ValueError:
+        If given location can not be found by the geolocator.
```

Comment:
This probably also raises an `ValueError` if `scheme` is invalid.

Reply:
yeah but that's taken care of from `extract_point` - that one raises its own ValueError, I checked

Reply:
I know, but might make sense to add it here since people might not immediately connect the two preprocessors. However, up to you.

Reply:
sure I added a ValuError here if not supplied, before extract point catches it, but if supplied and not either linear or nearest then extract point catches it

Reply:
I was just referring to the documentation, I didn't want you to add a check in the code :grimacing:

Reply:
```suggestion
    ValueError:
        If given location cannot be found by the geolocator.
```